x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_initializer.tsx

```diff
@@ -0,0 +1,292 @@
+  );
+  const [esqlQuery, setQuery] = useState<AggregateQuery>(initialInput?.query ?? defaultESQLQuery);
+  const isEsqlEnabled = useMemo(() => uiSettings.get(ENABLE_ESQL), [uiSettings]);
+
+  useEffect(() => {
+    if (initialInput?.viewType === undefined) {
+      // By default, if ES|QL is enabled, then use ES|QL
+      setViewType(
+        isEsqlEnabled ? FieldStatsInitializerViewType.ESQL : FieldStatsInitializerViewType.DATA_VIEW
+      );
+    }
+  }, [isEsqlEnabled, initialInput?.viewType]);
+
+  const isEsqlMode = viewType === FieldStatsInitializerViewType.ESQL;
+  const updatedProps = useMemo(() => {
+    return {
+      viewType,
+      title: initialInput?.title ?? defaultTitle,
+      dataViewId: isEsqlMode ? undefined : dataViewId,
+      query: isEsqlMode ? esqlQuery : undefined,
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dataViewId, viewType, esqlQuery.esql, isEsqlMode]);
+  const onESQLQuerySubmit = useCallback(
+    async (query: AggregateQuery, abortController: AbortController) => {
+      await onPreview({
+        viewType,
+        dataViewId: undefined,
+        query,
+      });
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [isEsqlMode]
+  );
+  const isEsqlFormValid = isEsqlMode ? isEsqlEnabled && isESQLQuery(esqlQuery) : true;
+  const isDataViewFormValid =
+    viewType === FieldStatsInitializerViewType.DATA_VIEW ? dataViewId !== '' : true;
+
+  return (
+    <>
+      <EuiFlyoutHeader
+        hasBorder={true}
+        css={css`
+          pointer-events: auto;
+          background-color: ${euiThemeVars.euiColorEmptyShade};
+        `}
+        data-test-subj="editFlyoutHeader"
+      >
+        <EuiFlexGroup justifyContent="spaceBetween" alignItems="center" responsive={false}>
+          <EuiFlexItem grow={false}>
+            <EuiTitle size="xs" data-test-subj="inlineEditingFlyoutLabel">
+              <h2>
+                {isNewPanel
+                  ? i18n.translate(
+                      'xpack.dataVisualizer.fieldStatisticsDashboardPanel.config.createLable',
+                      {
+                        defaultMessage: 'Create field statistics',
+                      }
+                    )
+                  : i18n.translate(
+                      'xpack.dataVisualizer.fieldStatisticsDashboardPanel.config.editLabel',
+                      {
+                        defaultMessage: 'Edit field statistics',
+                      }
+                    )}{' '}
+                <EuiIconTip
+                  type="iInCircle"
+                  content={i18n.translate(
+                    'xpack.dataVisualizer.fieldStatisticsDashboardPanel.config.samplingTooltip',
+                    {
+                      defaultMessage:
+                        'Field statistics are calculated from a smaller subset of documents to increase performance, so some accuracy might be lost.',
```

Comment:
@lcawl Can you help check if this is a good tooltip message to indicate that the data is sampled/using a smaller subset of data?

Reply:
Or if you want to specifically mention random sampling, this is consistent with the message we use in the data visualizer page and the log pattern analysis page for example:

```
Field statistics uses the random sampling aggregation to increase performance, but some accuracy might be lost.
```